src/zabapgit_objects.prog.abap

```diff
@@ -1007,9 +1013,14 @@ CLASS lcl_objects_program IMPLEMENTATION.
         not_found        = 2
         permission_error = 3
         OTHERS           = 4.
-    IF sy-subrc = 2.
+
+    lv_subrc = sy-subrc.
+
+    SET LOCALE LANGUAGE lv_curr_lang.
```

Comment:
Are you saving lv_subrc in a variable because `SET LOCALE LANGUAGE` changes the subrc?

Reply:
yeah, perhaps just check SY-SUBRC first and then SET LOCALE after that

Reply:
SET LOCAL LANGUAGE does not actually change SY-SUBRC, so the following would work. However this might be quite confusing to read, that's why I introduced the local variable to make clear that the IF block refers to the return code of the function call.

`

    CALL FUNCTION 'RPY_PROGRAM_READ'
      ...
      EXCEPTIONS
        cancelled        = 1
        not_found        = 2
        permission_error = 3
        OTHERS           = 4.

    SET LOCALE LANGUAGE lv_curr_lang.

    IF sy-subrc = 2.
      RETURN.
    ELSEIF sy-subrc <> 0.
      lcx_exception=>raise( 'Error reading program' ).
    ENDIF.
`

Setting local after checking the return code would result in repeating the same coding three times:

`

    CALL FUNCTION 'RPY_PROGRAM_READ'
      ...
      EXCEPTIONS
        cancelled        = 1
        not_found        = 2
        permission_error = 3
        OTHERS           = 4.

    IF sy-subrc = 2.
      SET LOCALE LANGUAGE lv_curr_lang.
      RETURN.
    ELSEIF sy-subrc <> 0.
      SET LOCALE LANGUAGE lv_curr_lang.
      lcx_exception=>raise( 'Error reading program' ).
    ENDIF.

    SET LOCALE LANGUAGE lv_curr_lang.
`